jaeger-core/src/main/java/com/uber/jaeger/samplers/ProbabilisticSampler.java

```diff
@@ -33,25 +33,38 @@
 public class ProbabilisticSampler implements Sampler {
   public static final String TYPE = "probabilistic";
 
-  private final long positiveSamplingBoundary;
-  private final long negativeSamplingBoundary;
-  private final double samplingRate;
-  private final Map<String, Object> tags;
+  private final double DELTA = 0.00001;
+  private long positiveSamplingBoundary;
+  private long negativeSamplingBoundary;
+  private double samplingRate;
+  private Map<String, Object> tags;
```

Comment:
@yurishkuro Could you elaborate more on why you prefer these to be final?

Having these be mutable means that we have have fields in `GuaranteedThroughputSampler` and `PerOperationSampler` be final. They also lend themselves to elegant tests because we only need to verify that these samplers correctly call the `ProbabilisticSampler#update`.

I don't think having `update` always return a new instance is a good idea because they can simply use the constructor to create it. I feel that in that case, any custom behavior like short circuiting creation should lie with the caller. I particularly don't like the idea of having a `getOrCreate` endpoint because I feel its very prone to errors.